service/Core/Handlers/DependencyInjection.cs

```diff
@@ -103,8 +103,8 @@ public static IServiceCollection AddDefaultHandlersAsHostedServices(this IServic
         // Build generic type: HandlerAsAHostedService<THandler>
         Type handlerAsAHostedServiceTHandler = typeof(HandlerAsAHostedService<>).MakeGenericType(tHandler);
 
-        Func<IServiceProvider, IHostedService> implementationFactory =
-            serviceProvider => (IHostedService)ActivatorUtilities.CreateInstance(serviceProvider, handlerAsAHostedServiceTHandler, stepName);
+        IHostedService implementationFactory(IServiceProvider serviceProvider)
```

Comment:
why this change?

Reply:
Because trying to rebuild the solution (using **KernelMemory.sln**, `main` branch), I get this compilation error:

![image](https://github.com/microsoft/kernel-memory/assets/3522534/21ffdd1c-26bc-4b8a-bbe8-31080ed3e2d8)

And the message point me to this page: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0039. I just updated Visual Studio 2022 to v17.9.3, perhaps it is a new analyzer?

Reply:
looks like a new analyzer only in VS, odd